lib/stdlib/src/io_ansi.erl

```diff
@@ -0,0 +1,2099 @@
+<<"\e[34m">>
+```
+""".
+?SPEC(blue).
+?FUNCTION(blue).
+
+-doc """
+Change foreground (aka text) color to magenta.
+
+Example:
+```erlang
+1> io_ansi:magenta().
+<<"\e[35m">>
+```
+""".
+?SPEC(magenta).
+?FUNCTION(magenta).
+
+-doc """
+Change foreground (aka text) color to cyan.
+
+Example:
+```erlang
+1> io_ansi:cyan().
+<<"\e[36m">>
+```
+""".
+?SPEC(cyan).
+?FUNCTION(cyan).
+
+-doc """
+Change foreground (aka text) color to white.
+
+Example:
+```erlang
+1> io_ansi:white().
+<<"\e[37m">>
+```
+""".
+?SPEC(white).
+?FUNCTION(white).
+
+-doc """
+Change foreground (aka text) color to index color. `Index` 0-15 are equivilant to
+the named colors in `t:foreground_color/0` in the order that they are listed.
+
+Example:
+```erlang
+1> io_ansi:color(5).
+<<"\e[35m">>
+2> io_ansi:color(80).
+<<"\e[38;5;80m">>
+```
+""".
+-spec color(Index :: 0..255 | 0..87) -> unicode:chardata().
+?FUNCTION(color, Index).
+
+-doc """
+Change foreground (aka text) color to RGB color.
+
+Example:
+```erlang
+1> io_ansi:color(255, 0, 0).
+<<"\e[38;2;255;0;0m">>
+```
+""".
+-spec color(0..255, 0..255, 0..255) -> unicode:chardata().
+?FUNCTION(color, Red, Green, Blue).
+
+-doc """
+Change foreground (aka text) color to the default color.
+
+Example:
+```erlang
+1> io_ansi:default_color().
+<<"\e[39m">>
+```
+""".
+?SPEC(default_color).
+?FUNCTION(default_color).
+
+-doc """
+Change background color to black.
+
+Example:
+```erlang
+1> io_ansi:black_background().
+<<"\e[40m">>
+```
+""".
+?SPEC(black_background).
+?FUNCTION(black_background).
+
+-doc """
+Change background color to red.
+
+Example:
+```erlang
+1> io_ansi:red_background().
+<<"\e[41m">>
+```
+""".
+?SPEC(red_background).
+?FUNCTION(red_background).
+
+-doc """
+Change background color to green.
+
+Example:
+```erlang
+1> io_ansi:green_background().
+<<"\e[42m">>
+```
+""".
+?SPEC(green_background).
+?FUNCTION(green_background).
+
+-doc """
+Change background color to yellow.
+
+Example:
+```erlang
+1> io_ansi:yellow_background().
+<<"\e[43m">>
+```
+""".
+?SPEC(yellow_background).
+?FUNCTION(yellow_background).
+
+-doc """
+Change background color to blue.
+
+Example:
+```erlang
+1> io_ansi:blue_background().
+<<"\e[44m">>
+```
+""".
+?SPEC(blue_background).
+?FUNCTION(blue_background).
+
+-doc """
+Change background color to magenta.
+
+Example:
+```erlang
+1> io_ansi:magenta_background().
+<<"\e[45m">>
+```
+""".
+?SPEC(magenta_background).
+?FUNCTION(magenta_background).
+
+-doc """
+Change background color to cyan.
+
+Example:
+```erlang
+1> io_ansi:cyan_background().
+<<"\e[46m">>
+```
+""".
+?SPEC(cyan_background).
+?FUNCTION(cyan_background).
+
+-doc """
+Change background color to white.
+
+Example:
+```erlang
+1> io_ansi:white_background().
+<<"\e[47m">>
+```
+""".
+?SPEC(white_background).
+?FUNCTION(white_background).
+
+-doc """
+Change background color to index color. `Index` 0-15 are equivilant to
+the named colors in `t:background_color/0` in the order that they are listed.
+
```

Comment:
I guess you have already thought about this but why not have:
`foreground/1` and `background/1` accept color atoms directly?  
Then you could remove all Color()  and Color_background() functions.